### ISSUE TITLE
VHD files left over in Primary Storage

### ISSUE THREAD

**Issue by nxsbi** [apache/cloudstack#6482]
##### ISSUE TYPE

Bug Report

##### COMPONENT NAME

~~~
Primary Storage
~~~

##### CLOUDSTACK VERSION

~~~
4.15
~~~

##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

CentOS 7

##### SUMMARY

We need to decommission a Primary Storage due to end of life of hardware. I have setup new Primary Storage, and though UI moved all disks to new storage (one by one). After everything is moved over, we still have a bunch of VHD files left in the old Primary Storage. I am not able to find any matches to the UUID/Path in the VM_INSTANCES table. So I am not sure where they came from, and if I were to delete them or disconnect the Primary Storage if anything will break

##### STEPS TO REPRODUCE

~~~
N/A
~~~

##### EXPECTED RESULTS

~~~
All VHD files were moved out of the old Primary Storage when moving a Volume to new primary storage
~~~

##### ACTUAL RESULTS

~~~
Multiple VHD Files are still present in old Primary Storage
~~~

**Comment by DaanHoogland** [comment]
@nxsbi did you try calling the [deleteStoragePool API](https://cloudstack.apache.org/api/apidocs-4.17/apis/deleteStoragePool.html) ?

**Comment by nxsbi** [comment]
Thanks @DaanHoogland 
I did not do that, but I placed the Storage Pool in Maintenance Mode (from GUI)
That actually moved a few remaining VHD that were in use by Cloudstack (SSVM and couple of virtual router)
In 4.15 there is no GUI option to move the storage of system VM or routers to another storage - it just gets randomly assigned.

So once it was in maintenance mode, I was able to move all other VHD files into another folder (just in case something breaks). But everything worked fine. 
Iwill be able to decomm the old storage now (but will wait a few days - just in case :-)
